### ISSUE TITLE
Kivy build failed

### ISSUE THREAD

**Issue by c0ff330k** [kivy/python-for-android#2735]
**Software Versions**
* Python: 3.9.9
* OS: Xubuntu 22.04 LTS
* Kivy: github master branch
* Kivy installation method: Buildozer

**Describe the bug**
Kivy build failed

**Code and Logs and screenshots**
```

  Environ change use_sdl2 -> True
  Current directory is: /tmp/pip-req-build-xg81mlga
  Source and initial build directory is:
  Python path is:
  /tmp/pip-req-build-xg81mlga
  /home/c0ff330k/app/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process
  /tmp/pip-build-env-1726v52_/site
  /usr/local/lib/python39.zip
  /home/c0ff330k/app/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/other_builds/hostpython3/desktop/hostpython3/Lib
  /home/c0ff330k/app/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/other_builds/hostpython3/desktop/hostpython3/native-build/build/lib.linux-x86_64-3.9
  /tmp/pip-build-env-1726v52_/overlay/lib/python3.9/site-packages
  /tmp/pip-build-env-1726v52_/normal/lib/python3.9/site-packages
  /tmp/pip-req-build-xg81mlga/kivy/modules
  /home/c0ff330k/.kivy/mods


  Found Cython at /tmp/pip-build-env-1726v52_/overlay/lib/python3.9/site-packages/Cython/__init__.py
  Detected supported Cython version 0.29.32
  Using this graphics system: OpenGL
  WARNING: A problem occurred while running pkg-config --libs --cflags gstreamer-1.0 (code 1)

  b"Package gstreamer-1.0 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `gstreamer-1.0.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'gstreamer-1.0' found\n"

  WARNING: A problem occurred while running pkg-config --libs --cflags sdl2 SDL2_ttf SDL2_image SDL2_mixer (code 1)

  b"Package sdl2 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `sdl2.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'sdl2' found\nPackage SDL2_ttf was not found in the pkg-config search path.\nPerhaps you should add the directory containing `SDL2_ttf.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'SDL2_ttf' found\nPackage SDL2_image was not found in the pkg-config search path.\nPerhaps you should add the directory containing `SDL2_image.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'SDL2_image' found\nPackage SDL2_mixer was not found in the pkg-config search path.\nPerhaps you should add the directory containing `SDL2_mixer.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'SDL2_mixer' found\n"

  SDL2: missing sub library SDL
  SDL2: missing sub library SDL_mixer
  SDL2: missing sub library SDL_ttf
  SDL2: missing sub library SDL_image
  WARNING: A problem occurred while running pkg-config --libs --cflags pangoft2 (code 1)

  b"Package pangoft2 was not found in the pkg-config search path.\nPerhaps you should add the directory containing `pangoft2.pc'\nto the PKG_CONFIG_PATH environment variable\nNo package 'pangoft2' found\n"

  ERROR: Dependency for context.pyx not resolved: config.pxi
  ERROR: Dependency for compiler.pyx not resolved: config.pxi
  ERROR: Dependency for context_instructions.pyx not resolved: config.pxi
  ERROR: Dependency for fbo.pyx not resolved: config.pxi
  ERROR: Dependency for gl_instructions.pyx not resolved: config.pxi
  ERROR: Dependency for instructions.pyx not resolved: config.pxi
  ERROR: Dependency for opengl.pyx not resolved: config.pxi
  ERROR: Dependency for opengl_utils.pyx not resolved: config.pxi
  ERROR: Dependency for shader.pyx not resolved: config.pxi
  ERROR: Dependency for stencil_instructions.pyx not resolved: config.pxi
  ERROR: Dependency for scissor_instructions.pyx not resolved: config.pxi
  ERROR: Dependency for texture.pyx not resolved: config.pxi
  ERROR: Dependency for vbo.pyx not resolved: config.pxi
  ERROR: Dependency for vertex.pyx not resolved: config.pxi
  ERROR: Dependency for vertex_instructions.pyx not resolved: config.pxi
  ERROR: Dependency for cgl.pyx not resolved: config.pxi
  ERROR: Dependency for cgl_mock.pyx not resolved: config.pxi
  ERROR: Dependency for cgl_gl.pyx not resolved: config.pxi
  ERROR: Dependency for cgl_glew.pyx not resolved: config.pxi
  ERROR: Dependency for cgl_sdl2.pyx not resolved: config.pxi
  ERROR: Dependency for svg.pyx not resolved: config.pxi
  ERROR: Dependency for boxshadow.pyx not resolved: config.pxi
  Building extensions in parallel using 2 cores
  Updated build directory to: build/lib.linux-x86_64-cpython-39
  Build configuration is:
   * use_rpi = 0
   * use_egl = 0
   * use_opengl_es2 = 0
   * use_opengl_mock = 0
   * use_sdl2 = 0
   * use_pangoft2 = 0
   * use_ios = 0
   * use_android = 0
   * use_mesagl = 0
   * use_x11 = 0
   * use_wayland = 0
   * use_gstreamer = 0
   * use_avfoundation = 0
   * use_osx_frameworks = 0
   * debug_gl = 0
   * kivy_sdl_gl_alpha_size = 8
   * debug = False
  Updated build/lib.linux-x86_64-cpython-39/kivy/include/config.h
  Updated kivy/include/config.h
  Updated build/lib.linux-x86_64-cpython-39/kivy/include/config.pxi
  Updated kivy/include/config.pxi
  Updated build/lib.linux-x86_64-cpython-39/kivy/setupconfig.py
  Updated kivy/setupconfig.py
  Detected compiler is unix
   error: command '/home/c0ff330k/.buildozer/android/platform/android-ndk-r25b/toolchains/llvm/prebuilt/linux-x86_64/bin/clang' failed with exit code 1
  error: subprocess-exited-with-error
  
  × Building wheel for Kivy (pyproject.toml) did not run successfully.
  \u2502 exit code: 1
  \u2570\u2500> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/c0ff330k/app/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/venv/bin/python /home/c0ff330k/app/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a/build/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpl0uos2f2
  cwd: /tmp/pip-req-build-xg81mlga
  Building wheel for Kivy (pyproject.toml) ... error
  ERROR: Failed building wheel for Kivy
Failed to build Kivy
ERROR: Could not build wheels for Kivy, which is required to install pyproject.toml-based projects


  STDERR:

# Command failed: /usr/bin/python3 -m pythonforandroid.toolchain create --dist_name=test_kt_app --bootstrap=sdl2 --requirements=python3,android,asynckivy,asyncgui,chardet,ntplib,pyjnius,pillow,ffpyplayer,openssl,ffpyplayer_codecs,certifi,charset-normalizer,idna,requests,urllib3,https://github.com/c0ff330k/kivy/archive/master.zip --arch arm64-v8a --arch armeabi-v7a --copy-libs --color=always --storage-dir="/home/c0ff330k/app/.buildozer/android/platform/build-arm64-v8a_armeabi-v7a" --ndk-api=21 --ignore-setup-py --debug
# ENVIRONMENT:
#     SHELL = '/bin/bash'
#     SESSION_MANAGER = 'local/c0ff330k:@/tmp/.ICE-unix/1184,unix/c0ff330k:/tmp/.ICE-unix/1184'
#     WINDOWID = '77594627'
#     QT_ACCESSIBILITY = '1'
#     COLORTERM = 'truecolor'
#     XDG_CONFIG_DIRS = '/etc/xdg/xdg-xubuntu:/etc/xdg:/etc/xdg'
#     XDG_SESSION_PATH = '/org/freedesktop/DisplayManager/Session0'
#     XDG_MENU_PREFIX = 'xfce-'
#     CLUTTER_BACKEND = 'x11'
#     LANGUAGE = 'en_US:'
#     LC_ADDRESS = 'kk_KZ.UTF-8'
#     LC_NAME = 'kk_KZ.UTF-8'
#     SSH_AUTH_SOCK = '/run/user/1000/keyring/ssh'
#     USE_SDL2 = '1'
#     DESKTOP_SESSION = 'xubuntu'
#     LC_MONETARY = 'kk_KZ.UTF-8'
#     XDG_SEAT = 'seat0'
#     PWD = '/home/c0ff330k/app'
#     LOGNAME = 'c0ff330k'
#     XDG_SESSION_DESKTOP = 'xubuntu'
#     QT_QPA_PLATFORMTHEME = 'gtk2'
#     XDG_SESSION_TYPE = 'x11'
#     GPG_AGENT_INFO = '/run/user/1000/gnupg/S.gpg-agent:0:1'
#     XAUTHORITY = '/home/c0ff330k/.Xauthority'
#     XDG_GREETER_DATA_DIR = '/var/lib/lightdm-data/c0ff330k'
#     GDM_LANG = 'en_US'
#     HOME = '/home/c0ff330k'
#     LC_PAPER = 'kk_KZ.UTF-8'
#     LANG = 'en_US'
#     LS_COLORS = 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:'
#     XDG_CURRENT_DESKTOP = 'XFCE'
#     VTE_VERSION = '6800'
#     XDG_SEAT_PATH = '/org/freedesktop/DisplayManager/Seat0'
#     LESSCLOSE = '/usr/bin/lesspipe %s %s'
#     XDG_SESSION_CLASS = 'user'
#     TERM = 'xterm-256color'
#     LC_IDENTIFICATION = 'kk_KZ.UTF-8'
#     GTK_OVERLAY_SCROLLING = '0'
#     LESSOPEN = '| /usr/bin/lesspipe %s'
#     USER = 'c0ff330k'
#     DISPLAY = ':0.0'
#     SHLVL = '1'
#     LC_TELEPHONE = 'kk_KZ.UTF-8'
#     LC_MEASUREMENT = 'kk_KZ.UTF-8'
#     XDG_VTNR = '7'
#     XDG_SESSION_ID = 'c2'
#     XDG_RUNTIME_DIR = '/run/user/1000'
#     LC_TIME = 'kk_KZ.UTF-8'
#     XDG_DATA_DIRS = '/usr/share/xubuntu:/usr/share/xfce4:/usr/local/share:/usr/share:/var/lib/snapd/desktop:/usr/share'
#     PATH = '/home/c0ff330k/.buildozer/android/platform/apache-ant-1.9.4/bin:/home/c0ff330k/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/c0ff330k/.local/bin/'
#     GDMSESSION = 'xubuntu'
#     DBUS_SESSION_BUS_ADDRESS = 'unix:path=/run/user/1000/bus'
#     LC_NUMERIC = 'kk_KZ.UTF-8'
#     _ = '/home/c0ff330k/.local/bin/buildozer'
#     PACKAGES_PATH = '/home/c0ff330k/.buildozer/android/packages'
#     ANDROIDSDK = '/home/c0ff330k/.buildozer/android/platform/android-sdk'
#     ANDROIDNDK = '/home/c0ff330k/.buildozer/android/platform/android-ndk-r25b'
#     ANDROIDAPI = '33'
#     ANDROIDMINAPI = '21'
# 
# Buildozer failed to execute the last command
# The error might be hidden in the log above this error
# Please read the full log, and search for it before
# raising an issue with buildozer itself.
# In case of a bug report, please add a full log with log_level = 2

```

**Comment by misl6** [comment]
Hi @c0ff330k !

I've moved the issue to `python-for-android`, can you please post the full log and your `buildozer.spec` file?

**Comment by RobertFlatt** [comment]
> [INFO   ] [Kivy        ] v2.2.0.dev0

> https://github.com/c0ff330k/kivy/archive/master.zip

This is a third party fork of Kivy, probably best to ask in that proect.

Alternatively use the release version of Kivy (2.1.0) and if you use KivyMD, whatever works with the released Kivy.
I think that is KivyMD 1.1 but I am not certain.

**Comment by HyTurtle** [comment]
Well you are circumventing the kivy recipe, you can use `requirements.source.kivy` to set it to your local cloned copy or customise the recipe

**Comment by RobertFlatt** [comment]
Hey @c0ff330k OK, but there is no way for anybody else to know that. Anyway, what works is to avoid both.

KivyMD 1.2.0 uses a feature that is only available in the development version of Kivy 2.2.0.dev0  (there is no Kivy 2.2.0 despite what its says in the KivyMD documentation). This Kivy feature  (BoxShadow) has portability issues - it is not reliable, but Kivy 2.2.0.dev0 is development code so you might reasonably expect that. Previous to this KivMD tried to develop the same functionality and it too had portability issues, so this is clearly a challenging feature.

At this time it is unclear how this will resolve, use KivyMD 1.0.2, avoid newer versions.
